express/src/controllers/clients.ts

```diff
@@ -451,23 +457,24 @@ export const processRemoveRedirectUriFrom: RequestHandler = async (req, res) =>
     }
 };
 
-export const showChangeUserAttributesForm: RequestHandler = (req, res) => {
-    res.render("clients/change-user-attributes.njk", {
-        selectedUserAttributes: req.query.userAttributes?.toString().split(" "),
+export const showChangeScopesForm: RequestHandler = (req: Request, res: Response): void => {
+    res.render("clients/change-scopes.njk", {
+        selectedScopes: req.query.scopes?.toString().split(" "),
         serviceId: req.context.serviceId,
         selfServiceClientId: req.params.selfServiceClientId,
         clientId: req.params.clientId
     });
 };
 
-export const processChangeUserAttributesForm: RequestHandler = async (req, res) => {
+// eslint-disable-function @typescript-eslint/no-misused-promises
```

Comment:
We can avoid disabling the lint here. I suggest to try to remove the eslint disable line and try  : 

```
export const processChangeScopesForm: RequestHandler = async (req: Request, res: Response): Promise<void> => {
    try {
        const s4: SelfServiceServicesService = req.app.get("backing-service");
        const attributes = ["openid"];
        if (Array.isArray(req.body.scopes)) {
            attributes.push(...req.body.scopes);
        } else if (typeof req.body.scopes === "string") {
            attributes.push(req.body.scopes);
        }

        await s4.updateClient(
            nonNull(req.context.serviceId),
            req.params.selfServiceClientId,
            req.params.clientId,
            { scopes: attributes },
            nonNull(req.session.authenticationResult?.AccessToken)
        );

        req.session.updatedField = "scopes";
        res.redirect(`/services/${req.context.serviceId}/clients`);
    } catch (error) {
        console.error("Error processing change scopes form:", error);
        res.status(500).send("Internal Server Error");
    }
};
```

Reply:
Thanks I'll give that a go. This disable doesn't work anyway: the web sonar ignores it.

Reply:
I'm afraid that suggestion doesn't work either. It still throws it out.